src/molusce/moluscedialog.py

```diff
@@ -1897,10 +2092,14 @@ def __addRasterToCanvas(self, filePath):
         else:
             self.logMessage(self.tr("Can't load raster %s") % (filePath))
 
-    def __bandCount(self):
+    def __bandCount(self, sim=False):
         bands = 0
-        for _k, v in self.inputs["factors"].items():
-            bands += v.getBandsCount()
+        if sim == False:
```

Comment:
Can the variable take values ​​other than bool? If not, then it is better to replace it with `not sim`